src/mjolnir/shortcutbuilder.cc

```diff
@@ -81,21 +81,25 @@ bool EdgesMatch(const graph_tile_ptr& tile, const DirectedEdge* edge1, const Dir
   // TODO - should allow near matches?
   std::vector<std::string> edge1names = tile->GetNames(edge1);
   std::vector<std::string> edge2names = tile->GetNames(edge2);
-  if (edge1names.size() != edge2names.size()) {
+  std::sort(edge1names.begin(), edge1names.end());
+  std::sort(edge2names.begin(), edge2names.end());
+  if (edge1names != edge2names)
     return false;
-  }
-  for (const auto& name1 : edge1names) {
-    bool found = false;
-    for (const auto& name2 : edge2names) {
-      if (name1 == name2) {
-        found = true;
-        break;
-      }
-    }
-    if (!found) {
+
+  // if they have access restrictions those must match (for modes that use shortcuts)
+  if (edge1->access_restriction()) {
+    auto res1 = tile->GetAccessRestrictions(edge1 - tile->directededge(0), kVehicularAccess);
+    auto res2 = tile->GetAccessRestrictions(edge2 - tile->directededge(0), kVehicularAccess);
+    if (res1.size() != res2.size())
       return false;
+    for (size_t i = 0; i < res1.size(); ++i) {
+      res1[i].set_edgeindex(0);
+      res2[i].set_edgeindex(0);
```

Comment:
just one last question: why these last 2 lines? they're modifying a copy of the restriction and seems irrelevant for the equal comparison. I'd get rid of them if that's the case.

Reply:
restrictions dont have an equality operator implemented, so i thought i'll just compare using `.value()` rather than implement one. i cant just compare with `.value()` though because for two different edges, the edge indices will always be different, which means we will always return false below. by zeroing out the edge index of both edges' restrictions we can make sure to compare the only part we care about: the actual restriction meta data.

tldr we set them both to the same value exactly so that they dont matter in the comparison below (ie that part of the bitfield we dont care about is always matching)

Reply:
it does seem to me that the `edgeindex` is not relevant for the `value()` equality comparison, but maybe I'm blind to some concrete detail here? fwiw, the relevant tests pass if I remove those lines.

Reply:
I don't wanna be pedantic, but those things tend to confuse future-me and lead down to the same investigation I did now. I'd leave a comment once I understand why it's needed

Reply:
if the test still passes then id be skeptical its working properly. i was going to ask you to prove it worked by removing the tagging for the restrictions from those middle ways and proving that one long shortcut is made over the whole thing. if that is the case then i cannot see how these lines would not be needed.

Reply:
yes, I got also skeptical since codecov reports that it never got into the `return false` below. will add a quick test

Reply:
oh man i misread the accessrestriction header! in fact we need to change this comparison, sorry!

Reply:
pushed an update, sorry for the confusion